### DISCUSSION TITLE
How to continue to use pydantic_encoder and parse_datetime in a correct V2 version

### DISCUSSION THREAD

**Discussion by ponponon** [pydantic/pydantic#8167] [Pydantic V2]
How to continue to use pydantic_encoder and parse_datetime in correct V2 version 

In v1, we can handle dates and json related content in the following way

```python
from pydantic.datetime_parse import parse_datetime
from pydantic.json import pydantic_encoder
```

But in the v2 version, all of this has been removed!

But pydantic_encoder and parse_datetime are very basic and common functions, more important than anything else!

So, in the v2 era, I'm lost and don't know what to do.

What is the official pydantic plan for pydantic_encoder and parse_datetime? How do I continue to use them? Have they been moved to another package?

Should I continue to import them from v1, or use another package? If the latter, which package should I use?

**Comment by samuelcolvin** [comment]
That functionality has been replaced and the methods removed.

We support all the same functionality though. What are you trying to do?

**Reply by ponponon** [reply]
I want to continue using parse_datetime and pydantic_encoder, what are their alternatives? How do I import the alternatives?

**Comment by samuelcolvin** [comment]
Okay, I'll try one more time.

Date time validation is implemented in rust, it's not exported as a function, so you'd need to build a TypeAdapter and call that to parse/validate a datetime.

Dumping, including to JSON is again implemented in rust, so methods aren't exported directly. Hence my question - how are you using it, so I could recommend the best alternative.
